scripts/list_outdated_dependencies.py

```diff
@@ -0,0 +1,272 @@
+
+            # Strip the specifier symbols from the front of the string if desired
+            if strip:
+                specified = specified.lstrip("<=>!~")
+
+            # append package name, specified version, and latest available version
+            if not requirement.specifier.contains(pypi_version):
+                outdated.append([requirement.name, specified, pypi_version])
+    outdated.sort()
+    return outdated
+
+
+def print_table(
+    rows: list[list[str]],
+    headers: list[str],
+    delimiter: str = " | ",
+):
+    """
+    List outdated dependencies in a formatted table.
+
+    Args:
+        `outdated`: A list of lists containing strings.
+        `headers`: A list containing the header strings for the table columns.
+    """
+    header_lengths = [len(header) for header in headers]
+
+    # Find the maximum length of each column
+    col_widths = [max(len(str(cell)) for cell in col) for col in zip(*rows)]
+
+    # Create a row format based on the maximum column widths
+    row_format = delimiter.join(
+        f"{{:<{max(width, header_len)}}}"
+        for width, header_len in zip(col_widths, header_lengths)
+    )
+
+    print("  " + row_format.format(*headers))
+    for dependency in rows:
+        print("  " + row_format.format(*dependency))
+
+
+def main(transitive: bool = False):
+    """Check capped dependencies for newer versions.
+
+    Examine `pyproject.toml` and `requirements-dev.in` for capped dependencies.
+    Make a call to PyPI to see if any newer versions exist.
+
+    Use `transitive` to show outdated transitive dependencies.
+    """
+    modified_pyproject: dict[str, Any] = get_modified_pyproject()
+    main_dependencies = get_main_deps_pyproject(modified_pyproject)
+    optional_dependencies = get_optional_deps_pyproject(modified_pyproject)
+    dev_dependencies = get_deps_dev()
+
+    outdated_main = get_outdated_deps(main_dependencies)
+    outdated_optional = get_outdated_deps(optional_dependencies)
+    outdated_dev = get_outdated_deps(dev_dependencies)
+
+    found_outdated = any([outdated_main, outdated_optional, outdated_dev])
+    headers = ["PACKAGE", "SPECIFIED", "AVAILABLE"]
+    if outdated_main:
+        location = PYPROJECT_TOML_PATH.name + " - dependencies"
+        cli.echo_failure(f"Outdated dependencies from {location}:")
+        print_table(outdated_main, headers)
+    if outdated_optional:
+        location = PYPROJECT_TOML_PATH.name + " - optional-dependencies"
+        cli.echo_failure(f"Outdated dependencies from {location}:")
+        print_table(outdated_optional, headers)
+    if outdated_dev:
+        cli.echo_failure(f"Outdated dependencies from {DEV_DEPS_PATH.name}:")
+        print_table(outdated_dev, headers)
+
+    if not found_outdated:
+        cli.echo_success("All top-level dependencies up to date.")
+
+    if transitive:
+        top_level: set[str] = set(
+            [item[0] for item in outdated_main + outdated_optional + outdated_dev]
+        )
```

Comment:
```suggestion
        top_level: set[str] = {item[0] for item in outdated_main + outdated_optional + outdated_dev}
```

or 
```suggestion
        top_level = {item.name for item in outdated_main + outdated_optional + outdated_dev}
```
when using named tuples

Reply:
Changed to use tuples in [1fa422f](https://github.com/ghga-de/microservice-repository-template/pull/165/commits/1fa422ff320d515afd4a244b1c551a4f133e7d97)